Update PHP tracer docs to include valgrind and strace info (#11148)
DataDog/documentation: content/en/tracing/setup_overview/setup/php.md

## Troubleshooting an application crash

> In the unusual event of an application crash caused by the PHP tracer, typically because of a segmentation fault, the best thing to do is obtain a valgrind trace or core dump and contact Datadog support.

Reviewer: ```suggestion
In the unusual event of an application crash caused by the PHP tracer, typically because of a segmentation fault, the best thing to do is obtain a core dump or a Valgrind trace and contact Datadog support.
```
Reordering to match the information below. If you want Valgrind listed first here, switch the sections around too.

1. Ensure you have a suitable `ulimit` set in your system. You can set it to unlimited: `ulimit -c unlimited`.
1. If your application runs in a Docker container, changes to `/proc/sys/*` have to be done to the host machine. Contact your system administrator to know the options available to you. If you are able to, try recreating the issue in your testing or staging environments.

### Valgrind trace

> Preferably, if you want to give us as many hints as possible, try running the application with valgrind (install it from your package manager). This approach will also work in an unprivileged container (unlike core dumps by default).

Reviewer: ```suggestion
To gain more details about the crash, run the application with Valgrind. Unlike default core dumps, this approach works in an unprivileged container.

Install Valgrind with your package manager. Run the application with Valgrind enough to generate a few requests (not generally in production).
```

Author: a "default core dump" feels wrong to read, would then just write _Unlike core dumps_.

>    - For CLI this is as simple as executing `USE_ZEND_ALLOC=0 valgrind -- php path/to/script.php`.

Reviewer: ```suggestion
For a CLI application, run:
{{< code-block lang=shell>}}
USE_ZEND_ALLOC=0 valgrind -- php path/to/script.php
{{< /code-block >}}
```

>    - When running php-fpm execute `USE_ZEND_ALLOC=0 valgrind --trace-children=yes -- php-fpm -F --fpm-config ...`.

Reviewer 2: Just thought of this question: what is the performance hit expected running php-fpm/apache through `valgrind`? I expect that it is not usable in production, correct? In this case should we had a note mentioning this?

Author: Yes, that's just for reproducing a couple requests, not flooding it.

Reviewer: ```suggestion
When running `php-fpm` run:
{{< code-block lang="shell" >}}
 USE_ZEND_ALLOC=0 valgrind --trace-children=yes -- php-fpm -F --fpm-config <MORE_OPTIONS>
 {{< /code-block >}}
```

Reviewer: I added a suggested wording in the paragraph above.

>    - When using apache, execute `(. /etc/apache2/envvars; USE_ZEND_ALLOC=0 valgrind --trace-children=yes -- apache2 -X)`.

Reviewer: ```suggestion
When using Apache, run:
{{< code-block lang="shell" >}}
(. /etc/apache2/envvars; USE_ZEND_ALLOC=0 valgrind --trace-children=yes -- apache2 -X)`
{{< /code-block >}}
```

### Obtaining a strace

> Some issues arise due to external influence, in that case it can be valuable to have a `strace` (install it from your package manager). When providing Datadog support a strace, please make sure to include the `-f` option, to follow child processes as well.

Reviewer: ```suggestion
Some issues are caused by external factors, so it can be valuable to have a `strace`. 

Install `strace` with your package manager. When generating a `strace` to send to Datadog Support, ensure you use the `-f` option to follow child processes.
```

Obtaining the strace for:

  - CLI: `strace -f php path/to/script.php`
  - apache: `(. /etc/apache2/envvars; strace -f apache2 -X)`
  - php-fpm: `strace -f php-fpm -F --fpm-config ...`

> 

Reviewer: For a CLI application, run:
{{< code-block lang="shell" >}}
strace -f php path/to/script.php
{{< /code-block >}}

For `php-fpm`, run:
{{< code-block lang="shell" >}}
strace -f php-fpm -F --fpm-config <CONFIG_OPTIONS>
{{< /code-block >}}

For Apache, run:
{{< code-block lang="shell" >}}
(. /etc/apache2/envvars; strace -f apache2 -X)
{{< /code-block >}}
